components/sysutils/znapzend/Makefile

```diff
@@ -39,38 +39,55 @@ include $(WS_MAKE_RULES)/prep.mk
 include $(WS_MAKE_RULES)/configure.mk
 include $(WS_MAKE_RULES)/ips.mk
 
-COMPONENT_PREP_ACTION = ( cd $(SOURCE_DIR) && autoreconf --force --install --verbose --make )
+COMPONENT_PREP_ACTION =	( cd $(SOURCE_DIR) && autoreconf --force --install --verbose --make )
 
 COMPONENT_PRE_CONFIGURE_ACTION =	( $(CLONEY) $(SOURCE_DIR) $(@D) )
 
-# NOTE: Code below assumes that BINDIR64 is one layer under BINDIR32
-# Othewise might user GNU ln -rsf ...
-COMPONENT_POST_INSTALL_ACTION =	( $(MKDIR) $(PROTO_DIR)/$(CONFIGURE_BINDIR.$(BITS)) \
-        && case "$(BITS)" in \
-            32) ORIGDIR="../share/$(COMPONENT_NAME)/$(MACH32)/bin" ;; \
-            64) ORIGDIR="../../share/$(COMPONENT_NAME)/$(MACH64)/bin" ;; \
-            *) echo "Unsupported BITS value: $(BITS)" >&2 ; exit 1 ;; \
-        esac \
-        && for BINFILE in znapzend znapzendzetup znapzendztats ; do \
-            ln -sf "$$ORIGDIR/$$BINFILE" "$(PROTO_DIR)/$(CONFIGURE_BINDIR.$(BITS))/" || exit ; \
-        done )
+# Preferred (build) version of gcc
+GCCLIBDIR  =	$(GCC_ROOT)/lib
+PKG_OPTIONS +=	-D GCC_RUNTIME_PKG="$(GCC_RUNTIME_PKG)"
+
+# Note: Building and installing for one bitness, so for PATH to have
+# the binaries, it should be linked straight in bin/ (no $BITS magic)
+# In the second loop we help runtime linker find libssp.so that ListUtil's
+# library refers to. Note: for us, suffices if it is 32-bit.
+ELFEDIT = /usr/bin/elfedit
+COMPONENT_POST_INSTALL_ACTION =	( $(MKDIR) $(PROTO_DIR)/$(CONFIGURE_PREFIX)/bin \
+        && ORIGDIR="../share/$(COMPONENT_NAME)/bin" \
+        && for BINFILE in znapzend znapzendzetup znapzendztatz ; do \
+            rm -f "$(PROTO_DIR)/$(CONFIGURE_PREFIX)/bin/$$BINFILE" || true ; \
+            ln -sf "$$ORIGDIR/$$BINFILE" "$(PROTO_DIR)/$(CONFIGURE_PREFIX)/bin/" || exit ; \
+        done \
+        && find $(PROTO_DIR)/$(CONFIGURE_PREFIX)/share/$(COMPONENT_NAME)/lib/ -type f -name '*.so*' \
+            | while read LIBFILE ; do \
+                $(CHMOD) +w "$$LIBFILE" \
+                && $(ELFEDIT) -e "dyn:rpath $(GCCLIBDIR)" "$$LIBFILE" \
+                && $(CHMOD) -w "$$LIBFILE" \
+                || exit ; \
+            done \
+        )
+
+# Enable ASLR for this component
+ASLR_MODE = $(ASLR_ENABLE)
```

Comment:
ASLR_MODE oi-userland macros is no-op on OI

Reply:
Fixed